lib/ansible/modules/unarchive.py

```diff
@@ -602,8 +601,13 @@ def is_unarchived(self):
             # Note: this timestamp calculation has a rounding error
             # somewhere... unzip and this timestamp can be one second off
             # When that happens, we report a change and re-unzip the file
-            dt_object = datetime.datetime(*(time.strptime(pcs[6], '%Y%m%d.%H%M%S')[0:6]))
-            timestamp = time.mktime(dt_object.timetuple())
+            DT_RE = re.compile(r'^(\d{4})(\d{2})(\d{2})\.(\d{2})(\d{2})(\d{2})$')
+            match = DT_RE.match(pcs[6])
+            if match:
+                timestamp = time.mktime(time.struct_time(int(m) for m in match.groups() + (0, 0, 0)))
+            else:
+                # Assume epoch date
+                timestamp = time.mktime(time.struct_time([1980, 1, 1, 0, 0, 0, 0, 0, 0]))
```

Comment:
I'm wondering if we should fall back to the epoch within the `if` statement too (if the timestamp is less than the epoch) since `struct_time` won't complain about an invalid date.